src/main/java/com/lambdaworks/redis/SimpleRedisConnection.java

```diff
@@ -0,0 +1,144 @@
+     * <br/>
+     * Sets field in the hash stored at key to value. If key does not exist, a new key holding a hash is created.
+     * <br/>
+     * If field already exists in the hash, it is overwritten.
+     *
+     * @param key
+     * @param field
+     * @param value
+     * @param <T>
+     * @return true if field is a new field in the hash and value was set.
+     * <br/> false if field already exists in the hash and the value was updated.
+     */
+    public <T> boolean cache(String key, String field, T value) {
+
+        RedisConnection<byte[], byte[]> redisCommands = null;
+        try {
+            redisCommands = getRedisConnection();
+
+            byte[] serializedValues = new JavaSerializer().serializeObject(value);
+            redisCommands.hset(key.getBytes(), field.getBytes(), serializedValues);
+
+            return true;
+        } catch (Exception e) {
+            logger.error("Exception Occurred while caching a value in redis", e);
+        } finally {
+            closeConnection(redisCommands);
+        }
+        return false;
+    }
+
+    /**
+     * Get a generic cached value from redis
+     * Returns the value associated with field in the hash stored at key.
+     *
+     * @param key
+     * @param field
+     * @param <T>
+     * @return the value stored in a deserialized form or null when field is not present in the hash or key does not exist.
+     */
+    public <T> T get(String key, String field) {
+
+        RedisConnection<byte[], byte[]> redisCommands = null;
+        try {
+            redisCommands = getRedisConnection();
+
+            byte[] serializedData = redisCommands.hget(key.getBytes(), field.getBytes());
+            T deserializeObject = new JavaSerializer().deserializeObject(serializedData);
+
+            return deserializeObject;
+        } catch (Exception e) {
+            logger.error("Exception Occurred while getting a value from redis", e);
+        } finally {
+            closeConnection(redisCommands);
+        }
+        return null;
+    }
+
+    /**
+     * Clear a cached value in redis
+     * Removes the specified fields from the hash stored at key. Specified fields that do not exist within this hash are ignored.
+     *
+     * @param key
+     * @param fields
+     * @return the number of fields that were removed from the hash, not including specified but non existing fields.
+     */
+    public Long clear(String key, String... fields) {
+
+        RedisConnection<byte[], byte[]> redisCommands = null;
+        try {
+            redisCommands = getRedisConnection();
+
+            int maxBytes = Integer.MIN_VALUE;
+
+            for (int i = 0; i < fields.length; i++) {
+                if (fields[i].getBytes().length > maxBytes)
+                    maxBytes = fields[i].getBytes().length;
+            }
+
+            byte[][] fieldsInBytes = new byte[fields.length][maxBytes + 1];
```

Comment:
`new byte[fields.length][]` is sufficient because you initialize the second dimension in the loop below.